server/odc-service/src/main/java/com/oceanbase/odc/service/datatransfer/DataTransferService.java

```diff
@@ -244,28 +246,12 @@ public UploadFileResult getMetaInfo(@NonNull String fileName) throws IOException
         if (!uploadFile.exists() || !uploadFile.isFile()) {
             throw new IllegalArgumentException("Target is not a file or does not exist, " + fileName);
         }
+        String uploadFileName = uploadFile.getName();
 
         // If the file is from third party like PL/SQL, this will convert it compatible with ob-loader.
         ThirdPartyOutputConverter.convert(uploadFile);
 
-        String uploadFileName = uploadFile.getName();
-        if (StringUtils.endsWithIgnoreCase(uploadFileName, ".zip")) {
-            // 疑似 zip 压缩文件，需要进一步确认是否合法
-            try {
-                DumperOutput dumperOutput = new DumperOutput(uploadFile);
-                return UploadFileResult.ofDumperOutput(fileName, dumperOutput);
-            } catch (Exception e) {
-                log.warn("Not a valid zip file, file={}", fileName, e);
-                logger.warn("Not a valid zip file, file={}", fileName, e);
-                return UploadFileResult.ofFail(ErrorCodes.ImportInvalidFileType, new Object[] {uploadFileName});
-            }
-        } else if (StringUtils.endsWithIgnoreCase(uploadFileName, ".csv")) {
-            return UploadFileResult.ofCsv(fileName);
-        } else if (StringUtils.endsWithIgnoreCase(uploadFileName, ".sql")
-                || StringUtils.endsWithIgnoreCase(uploadFileName, ".txt")) {
-            return UploadFileResult.ofSql(fileName);
-        }
-        return UploadFileResult.ofFail(ErrorCodes.ImportInvalidFileType, new Object[] {uploadFileName});
+        return new OBMySQLDataTransferExtension().getImportFileInfo(fileName, uploadFile.toURI().toURL());
```

Comment:
can not new this extension directly, you should use PluginUtil to get the ob-mysql's instance of DataTransferExtension

Reply:
fixed.